### PR TITLE
Redux added

### PR DESCRIPTION
**Pull request by ishimwepati**
Hello !|

Below are the features implemented in this project : 

I added Redux Toolkit,
I created a directory that will contain all my Redux logic (/src/redux)
I configured a Redux store (/src/redux/store.js)
I defined a slice of state for books that:

      - Includes an array of books (initial state: empty array)
      - Includes a reducer that adds a book
      - Includes a reducer that removes a book

I defined a slice of state for categories that:

      - Includes an array of categories (initial state: empty array)
      - Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

I made sure my structure application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files.

Your suggestions are most welcome !

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.books.push(action.payload);
+    },
+    removeBook: (state, action) => {
+      state.books = state.books.filter((book) => book.id !== action.payload);
+    },
+  },
+});
```
Comment (VICTOR-CHUKWU) [inline src/redux/books/booksSlice.js:18]: - [ ] good implementation of reducer and actions

File: README.md [hunk 2]
```diff
@@ -100,6 +101,7 @@ Now in your termonal do this :
 > - First : "npm install" : to install all npm dependencies
 > - Then type "npm run build"
 > - npm install react-router-dom
+> - npm install @reduxjs/toolkit react-redux
```
Comment (VICTOR-CHUKWU) [inline README.md:104]: - [x] your MIT link is broken, to give your project proper license, kindly fix this

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @ishimwepati 👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed
- Good use of redux toolkit


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (ishimwepati) [reply README.md:104]: I have fixed it !

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (ishimwepati) [reply src/redux/books/booksSlice.js:18]: Thanks for your appreciation!

**Review by victorokoroji — APPROVED** [review]
Hi @ishimwepati , 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Requested changes implemented✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**